src/utils/narrow.js

```diff
@@ -123,17 +124,19 @@ export const pmNarrowFromUsersUnsafe = (recipients: $ReadOnlyArray<UserOrBot>):
 export const pm1to1NarrowFromUser = (user: UserOrBot): Narrow =>
   pmNarrowInternal(pmKeyRecipientsFor1to1(user.user_id));
 
-export const specialNarrow = (operand: string): Narrow => {
-  if (operand === 'starred') {
-    return Object.freeze({ type: 'starred' });
-  }
-  if (operand === 'mentioned') {
-    return Object.freeze({ type: 'mentioned' });
-  }
-  if (operand === 'private') {
-    return Object.freeze({ type: 'all-pm' });
+export const specialNarrow = (operand: 'starred' | 'mentioned' | 'dm' | 'private'): Narrow => {
+  switch (operand) {
+    case 'starred':
+      return Object.freeze({ type: 'starred' });
+    case 'mentioned':
+      return Object.freeze({ type: 'mentioned' });
+    case 'dm': // new in server-future; means the same as 'private'
```

Comment:
Lower in this file at `apiNarrowOfNarrow`, let's add a TODO-server comment like this:
```js
    // TODO(server-7.0): send `is:dm` rather than `is:private`
    allPrivate: () => [{ operator: 'is', operand: 'private' }],
```

And I guess another like this:
```js
      // TODO(server-7.0): send `dm` rather than `pm-with`
      // TODO(server-2.1): just send IDs instead
      return [{ operator: 'pm-with', operand: emails.join(',') }];
```